x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/hooks/use_fetch_agents_data.test.tsx

```diff
@@ -101,74 +104,124 @@ jest.mock('../../../../hooks', () => ({
   }),
 }));
 
-describe('useFetchAgentsData', () => {
-  const startServices = useStartServices();
-  const mockErrorToast = startServices.notifications.toasts.addError as jest.Mock;
-
-  beforeAll(() => {
-    mockedExperimentalFeaturesService.get.mockReturnValue({
-      displayAgentMetrics: true,
-    } as any);
-  });
-
-  beforeEach(() => {
-    mockErrorToast.mockReset();
-    mockErrorToast.mockResolvedValue({});
-  });
-
-  it('should fetch agents and agent policies data', async () => {
-    const renderer = createFleetTestRendererMock();
-    const { result } = renderer.renderHook(() => useFetchAgentsData());
-    await waitFor(() => new Promise((resolve) => resolve(null)));
-
-    expect(result?.current.selectedStatus).toEqual(['healthy', 'unhealthy', 'updating', 'offline']);
-    expect(result?.current.allAgentPolicies).toEqual([
-      {
-        id: 'agent-policy-1',
-        name: 'Agent policy 1',
-        namespace: 'default',
-      },
-      {
-        id: 'agent-policy-managed',
-        managed: true,
-        name: 'Managed Agent policy',
-        namespace: 'default',
-      },
-    ]);
-
-    expect(result?.current.agentPoliciesIndexedById).toEqual({
-      'agent-policy-1': {
-        id: 'agent-policy-1',
-        name: 'Agent policy 1',
-        namespace: 'default',
-      },
-    });
-    expect(result?.current.kuery).toEqual(
-      'status:online or (status:error or status:degraded) or (status:updating or status:unenrolling or status:enrolling) or status:offline'
-    );
-    expect(result?.current.currentRequestRef).toEqual({ current: 2 });
-    expect(result?.current.pagination).toEqual({ currentPage: 1, pageSize: 5 });
-    expect(result?.current.pageSizeOptions).toEqual([5, 20, 50]);
-  });
-
-  it('sync querystring kuery with current search', async () => {
-    const renderer = createFleetTestRendererMock();
-    const { result } = renderer.renderHook(() => useFetchAgentsData());
-
-    await waitFor(() => expect(renderer.history.location.search).toEqual(''));
-
-    // Set search
-    await act(async () => {
-      result.current.setSearch('active:true');
-    });
-
-    await waitFor(() => expect(renderer.history.location.search).toEqual('?kuery=active%3Atrue'));
-
-    // Clear search
-    await act(async () => {
-      result.current.setSearch('');
+function blocked(fn: any) {
```

Comment:
simple way to detect if the event loop is blocked, will call fn if the event loop is delayed for more than 1s